File: SpringServer/spring/src/main/java/com/springbackend/dto/UserDTO.java
package com.springbackend.dto;


import com.springbackend.entity.User;
import jakarta.persistence.Column;
import jakarta.persistence.Id;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor(force = true)
@AllArgsConstructor
public class UserDTO {
    private String userId;
    private String password;
    private Float height;
    private Float weight;
    private Float muscleMass;
    private Float fatMass;
    private String gender;
    private String shape;
    private Integer RM_bench;
    private Integer RM_squat;
    public UserDTO(User user) {
        this.userId = user.getUserId();
        this.password = user.getPassword();
        this.height = user.getHeight();
        this.weight = user.getWeight();
        this.muscleMass = user.getMuscleMass();
        this.shape = user.getShape();
        this.gender = user.getGender();
        this.fatMass = user.getFatMass();
        this.RM_bench = user.getRM_bench();
        this.RM_squat = user.getRM_squat();
    }




    public User toEntity() {
        return User.builder()
                .userId(userId)
                .password(password)
                .height(height)
                .weight(weight)
                .muscleMass(muscleMass)
                .fatMass(fatMass)
                .shape(shape)
                .gender(gender)
                .RM_bench(RM_bench)
                .RM_squat(RM_squat)
                .build();
    }


}

File: CapAndroid/android/app/src/main/java/com/team6/routineapp/DTO/UserDTO.kt
package com.team6.routineapp.dto

import com.google.gson.annotations.SerializedName

data class UserDTO(
    @SerializedName("userId")
    var userId: String? = null,

    @SerializedName("password")
    var password: String? = null,

    @SerializedName("height")
    var height: Float? = null,

    @SerializedName("weight")
    var weight: Float? = null,

    @SerializedName("muscleMass")
    var muscleMass: Float? = null,

    @SerializedName("fatMass")
    var fatMass: Float? = null,

    @SerializedName("gender")
    var gender: String? = null,

    @SerializedName("shape")
    var shape: String? = null,

    @SerializedName("rm_bench")
    var rm_bench: Int? = null,

    @SerializedName("rm_squat")
    var rm_squat: Int? = null
)